### ISSUE TITLE
Can't use Ember.inject.service() on Routes

### ISSUE THREAD

**Issue by jamesarosen** [emberjs/ember.js#11197]
When I do
```js
export default Ember.Route.extend({
  foo: Ember.inject.service(),

  model: function() {
    return this.foo.get('bar');
  }
});
```
I get
```
Error while processing route: translate Object.defineProperty called on non-object TypeError: Object.defineProperty called on non-object
    at Object.defineProperty (native)
    at Object.meta (http://localhost:7357/assets/vendor.js:27894:27)
    at Object.defineProperty (http://localhost:7357/assets/vendor.js:25698:20)
    at Descriptor.AliasedProperty_oneWaySet [as set] (http://localhost:7357/assets/vendor.js:20214:16)
    at afterModel (http://localhost:7357/assets/dummy.js:664:17)
    at applyHook (http://localhost:7357/assets/vendor.js:55667:32)
    at Object.HandlerInfo.runSharedModelHook (http://localhost:7357/assets/vendor.js:53668:22)
    at Object.HandlerInfo.runAfterModelHook (http://localhost:7357/assets/vendor.js:53651:21)
    at __exports__.bind (http://localhost:7357/assets/vendor.js:55536:19)
    at tryCatch (http://localhost:7357/assets/vendor.js:55993:16)
```

I tried turning `EXTEND_PROTOTYPES` on to no avail.

Related to #11172

**Comment by mixonic** [comment]
@jamesarosen services must be `get`. That is a funky error though, I don't believe it is the one I normally see. Can you confirm your failure with `get('foo.bar')`?

**Comment by jamesarosen** [comment]
That's probably it. When I use `registry.injection` or `app.inject`, it doesn't require `get`.

**Comment by mixonic** [comment]
@jamesarosen yeah I had a bunch of code that didn't use `get` and was bit myself. Looking forward to that get-less future :+1:

**Comment by alexmherrmann** [comment]
Sorry guys, what was the resolution here?

```javascript
  loginService: Ember.inject.service('login'),
  afterModel: function() {
    var lserv = this.get('loginService');
    if(this.lserv.isLoggedIn()) {
      // do something here
    }
  }
```

I'm pretty new to ember, how would I get access to the service from within AfterModel? @jamesarosen  seemed to indicate that I need to use `get`? Would that also work with a function like `isLoggedIn()` though?

**Comment by ashoksudani** [comment]
@jamesarosen,
I just came across this issue. While debugging I found a hell of difference between the way we access injected service in Ember route/controller or whatever class.
e.g.
My service was : reportS : Ember.inject.service()
and when I debug in console with and without get I found following difference : 
> this.reportS
   InjectedProperty {type: "service", name: undefined, isDescriptor: true, _dependentKeys: undefined, _suspended: undefined…}

>this.get('reportS')
   Class {__ember1456827225649: null, __ember_meta__: Meta} 

Difference is clear when we use get we get access of full class.
So if we access the service property like this.get('ServiceName').get('SeriviceProperty') then it will return proper result.

This is also going to affect computed properties in service. If you chance the service property using "set()" then it will trigger computed property otherwise not.
